pkg/config/polling_manager_test.go

```diff
@@ -146,6 +150,129 @@ func TestNewAsyncPollingProjectConfigManagerWithNullDatafile(t *testing.T) {
 	mockRequester.AssertExpectations(t)
 }
 
+func TestGetConfigForPollingConfigManagerWaitsForTimeoutForInvalidDatafile(t *testing.T) {
+	blockingTimeout := 50 * time.Millisecond
+	mockDatafile := []byte("NOT-VALID")
+	mockRequester := new(MockRequester)
+	mockRequester.On("Get", []utils.Header(nil)).Return(mockDatafile, http.Header{}, http.StatusOK, nil)
+
+	sdkKey := "test_sdk_key"
+
+	configManager := NewPollingProjectConfigManager(sdkKey, WithRequester(mockRequester), WithBlockingTimeout(blockingTimeout))
+	mockRequester.AssertExpectations(t)
+
+	start := time.Now()
+	expectedExecutionTime := start.Add(blockingTimeout)
+	// blocking api for specified time.
+	_, _ = configManager.GetConfig()
+	actualExecutionTime := start.Add(time.Since(start))
+	assert.WithinDuration(t, expectedExecutionTime, actualExecutionTime, 10*time.Millisecond)
```

Comment:
@yasirfolio3 @msohailhussain Following up on my earlier comment.

To limit flakiness I suggest we remove this `assert.WithinDuration`. Instead, we can implement a timeout on the total execution time of this test case, failing when it exceeds a pretty large duration, like 1-2 seconds. This lets us be resilient to varying execution time of `GetConfig`, while still ensuring that our test fails when the implementation of the timeout inside `GetConfig` is broken.